dfusion_rust_core/src/models/order.rs

```diff
@@ -47,6 +47,73 @@ impl Order {
             sell_amount,
         }
     }
+
+    /// Calculates the utility of an order based on a buy and sell price as well
+    /// as the executed buy amount.
+    ///
+    /// # Returns
+    ///
+    /// Returns `Some(U256)` utility result if the inputs were valid and it
+    /// produced a valid positive utility.
+    pub fn utility(
```

Comment:
Can we add a unit test for `utility` and `disregarded_utility`? Those could also include cases where this returns None.

Reply:
In the works :smiley:

Reply:
Added a few unit tests